bin/prepare_score_distribution.py

```diff
@@ -204,11 +208,26 @@ def main(args=None):
             print("\nInfo: final data", flush=True)
             data.info(verbose=False, memory_usage=print_mem)
 
+            # Assign prediction_scores to precomputed bins, group and sum up weights
+            data["prediction_score_bin"] = pd.cut(data["prediction_score"], bins=bin_edges)
+            data = (
+                data.groupby(["prediction_score_bin", "condition_name", "allele_id"])
+                .agg(
+                    agg_weight=("weight_sum", "sum"),
+                )
+                .reset_index()
+            )
+            # For the sake of simplicity, just use bin center for plotting (anyway smoothed for violin plots)
+            # (todo: think about more elegant way for visualization if necessary)
+            data["prediction_score_bin_center"] = data["prediction_score_bin"].apply(
+                lambda x: np.mean([x.left, x.right])
+            )
```

Comment:
I'm not sure how large the data frame is, at this moment, but it could be faster to use list comprehension instead of apply. If the df is rather small due to the bins it doesn't really matter.

Reply:
Yes, it's fairly small in this case. But good to keep in mind in general! But then I guess one could also do it somehow vectorized then, right?